packages/router/src/util.ts

```diff
@@ -197,6 +197,16 @@ export const validatePath = (path: string) => {
     throw new Error(`Route path contains spaces: "${path}"`)
   }
 
+  if (/\{(ref|key)(:|\})/.test(path)) {
```

Comment:
```suggestion
  if (/{(?:ref|key)(?::|})/.test(path)) {
```
Now that I'm back home again I could play around with the regex. 
There's nothing wrong with what you have, but this does the same thing perhaps a little bit more efficiently by using non-capturing groups. I also removed the escaping of characters that don't need it in the positions they have in this regex